clang/include/clang/DependencyScanning/DependencyScanningWorker.h

```diff
@@ -169,13 +173,21 @@ class DependencyScanningWorker {
   friend CompilerInstanceWithContext;
   std::unique_ptr<CompilerInstanceWithContext> CIWithContext;
 
-  /// Actually carries out the scan. If \c OverlayFS is provided, it must be
-  /// based on top of DepFS.
-  bool scanDependencies(
-      StringRef WorkingDirectory, const std::vector<std::string> &CommandLine,
-      DependencyConsumer &Consumer, DependencyActionController &Controller,
-      DiagnosticConsumer &DC,
-      IntrusiveRefCntPtr<llvm::vfs::FileSystem> OverlayFS = nullptr);
+  /// Private helper functions to actually carry out the scan. If \c OverlayFS
+  /// is provided, it must be based on top of DepFS.
+  bool scanDependencies(StringRef WorkingDirectory,
+                        const std::vector<std::string> &CommandLine,
+                        DependencyConsumer &Consumer,
+                        DependencyActionController &Controller,
+                        DiagnosticConsumer &DC,
+                        llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FS);
+
+  bool scanDependencies(StringRef WorkingDirectory,
+                        ArrayRef<std::vector<std::string>> CommandLine,
+                        DependencyConsumer &Consumer,
+                        DependencyActionController &Controller,
+                        DiagnosticsEngine &Diags,
+                        llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FS);
```

Comment:
Nit: these two function declarations look too similar to my eyes. Maybe we can do something along the following line:

```suggestion
  bool scanDependenciesWithDiagConsumer(
                        DiagnosticConsumer &DC,
                        StringRef WorkingDirectory,
                        const std::vector<std::string> &CommandLine,
                        DependencyConsumer &Consumer,
                        DependencyActionController &Controller,
                        llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FS);

  bool scanDependenciesWithDiagEngine(
                        DiagnosticsEngine &Diags,
                        StringRef WorkingDirectory,
                        ArrayRef<std::vector<std::string>> CommandLine,
                        DependencyConsumer &Consumer,
                        DependencyActionController &Controller,
                        llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FS);
```

Note the renamed functions, and the reordered parameter lists.